app/controllers/openid_application_controller.rb

```diff
@@ -12,82 +12,25 @@
 require 'jwt'
 
 class OpenidApplicationController < ApplicationController
+  include OAuthConcerns
   skip_before_action :verify_authenticity_token
   skip_after_action :set_csrf_header
   before_action :authenticate
   TOKEN_REGEX = /Bearer /.freeze
 
   private
 
-  def permit_scopes(scopes, actions: [])
```

Comment:
this was the most difficult part of all this, because a lot of this implementation is in the controller module controllers just extend from. unfortunately, this wouldnt work because grape api definitions have to extend a grape specific class. 

what this does is just move this methods out into a module (OAuthConcerns) that is then included in this file above and can then be included as "helpers" in grape endpoints below.